src/realm/group_writer.cpp

```diff
@@ -752,9 +846,50 @@ GroupWriter::FreeListElement GroupWriter::reserve_free_space(size_t size)
 {
     auto chunk = search_free_space_in_part_of_freelist(size);
     while (chunk == m_size_map.end()) {
-        // No free space, so we have to extend the file.
-        auto new_chunk = extend_free_space(size);
-        chunk = search_free_space_in_free_list_element(new_chunk, size);
+        if (!m_under_evacuation.empty()) {
```

Comment:
Is this overly sophisticated? Would it be acceptable/reasonable to just back out of the evacuation process completely at this point instead?

Reply:
Perhaps. But I would like the process to be somewhat adaptive to real life situations :-)